pysteps/tracking/tdating.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+pysteps.tracking.tdating
+========================
+
+Thunderstorm Detection and Tracking (DATing) module
+This module was implemented following the procedures used in the TRT Thunderstorms
+Radar Tracking algorithm (:cite:`TRT2004`) used operationally at MeteoSwiss.
+Modifications include advecting the identified thunderstorms with the optical flow
+obtained from pysteps, as well as additional options in the thresholding.
+
+References
+...............
+:cite:`TRT2004`
+
+@author: mfeldman
+
+.. autosummary::
+    :toctree: ../generated/
+
+    dating
+    tracking
+    advect
+    match
+    couple_track
+"""
+
+import copy
+import sys
+
+import numpy as np
+import scipy.ndimage as ndi
+
+import pysteps.feature.tstorm as tstorm_detect
+from pysteps import motion
+from pysteps.exceptions import MissingOptionalDependency
+
+try:
+    import skimage
+
+    SKIMAGE_IMPORTED = True
+except ImportError:
+    SKIMAGE_IMPORTED = False
+if SKIMAGE_IMPORTED == True:
+    import skimage.draw as skid
+    import skimage.feature as skif
+    import skimage.measure as skime
+    import skimage.morphology as skim
+    import skimage.segmentation as skis
+try:
+    import pandas as pd
+
+    PANDAS_IMPORTED = True
+except ImportError:
+    PANDAS_IMPORTED = False
+
+
+def dating(
+    input_video,
+    timelist,
+    mintrack=3,
+    cell_list=None,
+    label_list=None,
+    start=0,
+    minref=35,
+    maxref=48,
+    mindiff=6,
+    minsize=50,
+    minmax=41,
+    mindis=10,
+    dyn_thresh=False,
+):
+    """
+    This function performs the thunderstorm detection and tracking DATing.
+    It requires a 3-D input array that contains the temporal succession of the 2-D data
+    array of each timestep. On each timestep the detection is performed, the identified
+    objects are advected with a flow prediction and the advected objects are matched to
+    the newly identified objects of the next timestep.
+    The last portion re-arranges the data into tracks sorted by ID-number.
+
+    Parameters
+    ----------
+    input_video : array-like
+        Array of shape (t,m,n) containing input image, with t being the temporal
+        dimension and m,n the spatial dimensions. Nan values are ignored.
+    timelist : list
+        List of length t containing string of time and date of each (m,n) field.
+    mintrack : int, optional
+        minimum length of cell-track to be counted. The default is 3.
```

Comment:
What is the dimension, pixels?

Reply:
The dimension for mintrack is number of timesteps, I expanded the docstrings accordingly.